### ISSUE TITLE
DB upgrade from 3.5 to 4.0 on docker failing with error as "WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause"

### ISSUE THREAD

**Issue by rajiv83** [neo4j/docker-neo4j#229]
- Neo4j image tag being used -`neo4j:latest`
- The output of the `docker version` command

**docker version**
Client: Docker Engine - Community
 Version:           19.03.5
 API version:       1.40
 Go version:        go1.12.12
 Git commit:        633a0ea
 Built:             Wed Nov 13 07:22:34 2019
 OS/Arch:           darwin/amd64
 Experimental:      false

server: Docker Engine - Community
 engine:
  version:          19.03.5
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.12.12
  Git commit:       633a0ea
  Built:            Wed Nov 13 07:29:19 2019
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.2.10
  GitCommit:        b34a5c8af56e510852c35414db4c1f4fa6172339
 runc:
  Version:          1.0.0-rc8+dev
  GitCommit:        3e425f80a8c931f88e6d94a8c831b9d5aa481657
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
================================
- Operating system: Mac OS (Mojave)
- Steps to reproduce -- Performing neo4j database upgrade on docker from 3.5 to 4.o with persistent data volume mounted
- Expected behaviour -- The upgrade should complete
- Actual behaviour --- It terminates before starting the upgrade

docker run --name rummy  -p 7474:7474 -p 7687:7687 -v /data/db:/data  -e NEO4J_dbms_allow__upgrade=true neo4j:latest
Directories in use:
  home:         /var/lib/neo4j
  config:       /var/lib/neo4j/conf
  logs:         /logs
  plugins:      /var/lib/neo4j/plugins
  import:       /var/lib/neo4j/import
  data:         /var/lib/neo4j/data
  certificates: /var/lib/neo4j/certificates
  run:          /var/lib/neo4j/run
Starting Neo4j.
2020-02-11 13:54:23.976+0000 INFO  ======== Neo4j 4.0.0 ========
2020-02-11 13:54:23.992+0000 INFO  Starting...
2020-02-11 13:54:28.983+0000 INFO  [graph.db] Starting upgrade of database
2020-02-11 13:54:30.295+0000 INFO  [graph.db] Migrating Store files (1/6):

[neo4j_upgrade_error.log](https://github.com/neo4j/docker-neo4j/files/4186118/neo4j_upgrade_error.log)

**Comment by jennyowen** [comment]
@rajiv83 
what specific 3.5 version are you updating from? 
also what's the folder permissions on your host data folder and the subfolders within it?

**Comment by rajiv83** [comment]
@jennyowen 
Thanks for looking at it

My current DB version is 3.5.14 community (Neo4j Kernel)

The permission are getting auto acquired for datafiles to neo4j:neo4j after mounting them during docker initialization.

# ls -ld /data
drwxr-xr-x 8 neo4j neo4j 256 Feb 10 07:11 /data

# ls -l /data
total 38632704
drwxr-xr-x  7 neo4j neo4j         224 Feb 10 07:14 databases
drwxr-xr-x  3 neo4j neo4j          96 Feb 10 07:37 dbms
-rw-r--r--  1 root  root            0 Feb  9 13:28 store_lock
drwxr-xr-x 37 root  root         1184 Feb  9 13:28 system
drwxr-xr-x  4 root  root          128 Feb  9 13:33 transactions


# ls -l databases
total 0
drwxr-xr-x 42 neo4j neo4j 1344 Feb 11 14:41 graph.db
-rw-r--r--  1 neo4j neo4j    0 Feb  9 13:33 store_lock
drwxr-xr-x 37 neo4j neo4j 1184 Feb  9 13:33 system

**Comment by rajiv83** [comment]
Hi @jennyowen,

After increasing the docker memory, I was able to upgrade the DB successfully.

I am all set with this issue

Thanks
Rajiv

**Comment by rajiv83** [comment]
All set

**Comment by rajiv83** [comment]
I keep getting below error in the logs at regular frequency.

2020-02-12 08:16:22.213+0000 WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=432, gcTime=0, gcCount=0}
2020-02-12 08:19:57.670+0000 WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=361, gcTime=0, gcCount=0}
2020-02-12 08:21:26.882+0000 WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=257, gcTime=0, gcCount=0}
2020-02-12 08:26:15.100+0000 WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=318, gcTime=0, gcCount=0}
2020-02-12 08:35:43.737+0000 WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=169, gcTime=0, gcCount=0}
2020-02-12 08:51:43.400+0000 WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=267, gcTime=0, gcCount=0}
2020-02-12 10:02:22.893+0000 WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=190, gcTime=0, gcCount=0}
2020-02-12 10:02:31.676+0000 WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=144, gcTime=0, gcCount=0}
2020-02-12 10:10:26.649+0000 WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=154, gcTime=0, gcCount=0}
2020-02-12 10:20:50.659+0000 WARN [o.n.k.i.c.VmPauseMonitorComponent] Detected VM stop-the-world pause: {pauseTime=514, gcTime=0, gcCount=0}

**Comment by jennyowen** [comment]
@rajiv83 I think that might be a problem with neo4j server itself rather than the docker container. You could try increasing the `dbms.memory.heap.max_size` setting in the conf file. If that doesn't work could you raise this as a new issue against https://github.com/neo4j/neo4j ?
